src/reset-password/service.ts

```diff
@@ -49,6 +62,9 @@ export async function createToken(user: User): Promise<string> {
     return token;
 }
 
+/**
+ * Checks if the token provided is valid based on the time provided
```

Comment:
Not a very good description. There's no time provided to this function.

How about:

```
Checks if a 'password reset token' is valid, and returns the associated user.

This function only works once for every token. After calling this function, the token automatically gets deleted.
```